### PR TITLE
Add ops diff

### PR DESCRIPTION
**Pull request by Liguanlong123**
### PR Category
Operator

### Type of Change
New Feature

### Description
This operator is a diff operation based on Triton, supporting 1D, 2D, 3D data. It dynamically selects the optimal kernel based on the input dimension and shape (e.g., flattening, vectorization, 2D tiling, 3D optimization) and enhances memory access and parallel efficiency through vectorized loading, cache-friendly layout, and dynamic BLOCK/TILE parameters, thereby efficiently performing functions similar to torch.diff.

### Progress
- [ ] Initial implementation of the diff operator.

### Performance
Accuracy rate: n taken from 1-50 is accurate. 

准确率测试

Performance testing: on par with torch implementation in a series of test cases

性能测试1
性能测试2

### CODE REVIEW

**Comment by meinie0826** [comment]
Does this implementation only support 1D, 2D and 3D?

File: src/flag_gems/ops/diff.py [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+
+
+@triton.jit
+def diff_kernel_1d_vectorized(
+    input_ptr, output_ptr,
+    N, out_L,
+    BLOCK_SIZE: tl.constexpr,
+    VECTOR_WIDTH: tl.constexpr
+):
+    """1D向量化内核"""
```
Comment (meinie0826) [inline src/flag_gems/ops/diff.py:45]: It would be best to provide the annotations in English.

File: src/flag_gems/ops/diff.py [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+
+    if prepend is not None:
+        input = torch.cat([prepend, input], dim=dim)
+    if append is not None:
+        input = torch.cat([input, append], dim=dim)
+    
+    out = input
+
+    for _ in range(n):
+        out = _diff_once(out, dim)
```
Comment (StrongSpoon) [inline src/flag_gems/ops/diff.py:298]: I suggest implementing "diff" within a single kernel. For n greater than 1, the shape of the output tensor can be readily calculated, and the values of higher-order differences adhere to the pattern of Pascal's triangle.

**Comment by StrongSpoon** [comment]
I'm concerned that the performance may rapidly decline once n exceeds one.

File: src/flag_gems/ops/diff.py (hunk 2 above)
Comment (Liguanlong123) [reply src/flag_gems/ops/diff.py:298]: Okay, I will rewrite my code as you mentioned.

**Comment by Liguanlong123** [comment]
> I'm concerned that the performance may rapidly decline once n exceeds one.

I will test cases when n is greater than 1 after fixing the issues, aiming to maintain performance.

**Comment by bin913** [comment]
@Liguanlong123 Do you still keep track of this PR?
